src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -365,12 +527,12 @@ public void acceptRecord(final SAMRecord rec) {
 
                     for (final AlignmentBlock block : rec.getAlignmentBlocks()) {
                         final int end = CoordMath.getEnd(block.getReferenceStart(), block.getLength());
-                        for (int pos=block.getReferenceStart(); pos<=end; ++ pos) {
-                            if (pos >= target.getStart() && pos <= target.getEnd()) {
-                                ++this.metrics.ON_TARGET_BASES;
-                                if (mappedInPair) ++this.metrics.ON_TARGET_FROM_PAIR_BASES;
-                                coverage.addBase(pos - target.getStart());
-                            }
+                        for (int pos=block.getReferenceStart(), readPos = block.getReadStart();
```

Comment:
same comment as above here, seems strange to include deletions and softclips as bases that are "on target", no?